src/LogicTasks/Semantics/Prolog.hs

```diff
@@ -139,8 +160,20 @@ verifyQuiz PrologConfig{..}
 start :: (PrologLiteral, PrologClause)
 start = (PrologLiteral True "a" ["x"], mkPrologClause [])
 
+parser' :: Parser PrologClause -> Parser (PrologLiteral, PrologClause)
+parser' clauseParser = do
+  tokenSymbol "("
+  lit <- parser
+  tokenSymbol ","
+  clause <- clauseParser
+  tokenSymbol ")"
+  pure (lit, clause)
+
 partialGrade :: OutputCapable m => PrologInst -> Delayed (PrologLiteral, PrologClause) -> LangM m
-partialGrade inst = (partialGrade' inst `withDelayed` parser) displayParseError
+partialGrade inst = (partialGrade' inst `withDelayed` parser' clauseParser) (const complainAboutWrongNotation)
+  where clauseParser | usesSetNotation inst = prologClauseSetParser
+                     | otherwise            = prologClauseFormulaParser
```

Comment:
This second case could just be `parser`?

Reply:
Counterpoint: Further below in this file there is `displayClause = if usesSetNotation then showPrologClauseAsSet else show`, instead of naming an explicit `showPrologClauseAsFormula` function.

Symmetry suggests that `show` and `parser` go hand in hand, and thus the conditionals here should mirror each other.

Or, put differently, getting errors when the `Parse` instance changes might be a good thing, because then one would have to react, maybe realizing that one has accidentally changed the default `parser` without thinking of also changing the default `show` etc.